### ISSUE TITLE
Rails 4.0.0 date_select bug

### ISSUE THREAD

**Issue by tanraya** [rails/rails#25571]
Hi!
I got following error:

```
>> f.object.created_at
=> Mon, 20 Jun 2016 13:15:29 EDT -04:00
>> f.date_select :created_at
!! #<NoMethodError: undefined method `to_datetime' for 0:Fixnum>
```

This seemed a bug in date_select form helper. Does anyone experienced this bug?

### System configuration
**Rails version**: 4.0.0
**Ruby version**: ruby 2.3.1p112 (RVM)

**Comment by rafaelfranca** [comment]
Can you please provide a sample application that reproduces the error?

**Comment by mattbrictson** [comment]
Hi, ran into the same problem with 4.0.0. It also affects 4.0.1, 4.0.2, and 4.0.3. It seems to be fixed in 4.0.4.

Here is the test case I used to reproduce the bug in 4.0.0-4.0.3: https://gist.github.com/mattbrictson/fc4a94b01627769c01348bbbd691fbbf

**Comment by rafaelfranca** [comment]
Right. But if it is fixed in 4.0.4 so it is already fixed. Thank you for reproduction script.

**Comment by mattbrictson** [comment]
Yes, understood. I was just trying to help the OP by pointing to the version where it was fixed. Thanks for taking the time to review!
